### DISCUSSION TITLE
Cleaning up records created by AnyFixture

### DISCUSSION THREAD

**Discussion by Epigene** [test-prof/test-prof#308] [General]
Hi!

I'm using the latest v1.4.2 of the gem and trying to set up canonical records with AnyFixture. I'm moving the setup to `before(:suite)` and omitting the shared example, so it leaves me with this:

```rb
RSpec.configure do |config|
  config.before(:suite) do  
    TestProf::AnyFixture.register(:project) do      
      FactoryBot.create(:project)
    end
  end
end
```

The created Project is there at suite end and will interefere with repeat spec runs. Does AnyFixture have automagic cleanup mechanism, and I've mis-configured something or am I expected to clean things up myself?

**Comment by palkan** [comment]
> Does AnyFixture have automagic cleanup mechanism

Yes, it does.

You've probably haven't activated it. You must add `require "test_prof/recipes/rspec/any_fixture"` for that. Do you have that line in your config?

**Comment by Epigene** [comment]
I have these testprof lines in config. Would I have access to AnyFixture class/module without it?

```rb
require "test_prof/recipes/rspec/before_all" # provides the `before_all` helper
require "test_prof/recipes/rspec/let_it_be"  # provides the `let_it_be` helper
require "test_prof/recipes/rspec/any_fixture" # provides `TestProf::AnyFixture.register` helper
```

**Comment by Epigene** [comment]
Very weird, the created records keep remaining in DB.
I've tried:

1. Switching to `before(:all)` when setting up
2. calling `TestProf::AnyFixture.reset` manually
3. calling `TestProf::AnyFixture.clean` manually

**Comment by Epigene** [comment]
I think I got a clue, AnyFixture does indeed clean up just fine if I let the example finish. But if I quit from inside an example, its transaction is not closed for some reason and that interferes with cleanup. I've opened a discussion in rspec-rails. https://github.com/rspec/rspec-rails/discussions/2805
